Update/theme support color palette

webmandesign opened the pull request:
## Description

Allowing `editor-color-palette` theme support setup with a single array of colors. This way only one argument is passed to `add_theme_support( 'editor-color-palette', $arg )` instead of multiple arguments for each color.

Fising issue #6425

webmandesign committed:
Set `editor-color-palette` theme support as array

webmandesign committed:
Set `editor-color-palette` theme support as array

webmandesign committed:
Updating `editor-color-palette` setup info using single array argument

webmandesign committed:
Updating `editor-color-palette` setup info using single array argument

samikeijonen commented:
Makes sense to me.

webmandesign commented on docs/extensibility/theme-support.md:
````diff
@@ -8,7 +8,7 @@ To opt-in for one of these features, call `add_theme_support` in the `functions.
 ```php
 function mytheme_setup_theme_supported_features() {
-	add_theme_support( 'editor-color-palette',
+	add_theme_support( 'editor-color-palette', array(
````
I've just added the array wrapper there because of [@jorgefilipecosta's comment](https://github.com/WordPress/gutenberg/pull/5273#discussion_r183658734)

gziolo replied:
Can we map it to an ordered array before it gets exposed to JavaScript?

webmandesign replied:
@gziolo 
You mean like declaring the theme support with this:
`array(
	'color_name_1' => '#color1',
	'color_name_2' => '#color2',
)`
and then transform it in PHP to a JS required form of:
`array(
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
	),
)`
?

If so, this is certainly doable. But I'd like some more input on this, more opinions (maybe from @jorgefilipecosta)? You see, current associated array implementation is more self-explanatory in my opinion and could actually serve well in case of future extending of the palette color features.

jorgefilipecosta replied:
Hi,
Although I agree that it would be easier for themes to declare support using this syntax:
```
array(
	'color_name_1' => '#color1',
	'color_name_2' => '#color2',
)
````

I think it could limit future advances as the color value itself is a string and not an object we can expand.
For example, one possible improvement I was thinking about was to let themes define the contexts in which colors appear (e.g: allow some colors only for background).
In the current syntax, it is a matter of allowing an optimal property:

```
array(
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
		'context': => array( 'background' ),
	),
)
````

webmandesign replied:
Just like expected ;)

jasmussen committed:
Enhance theme colors (#6438)

It's thursday, and I'm AFK tomorrow for a long weekend! Danish holiday, yeees! For that reason, I treated myself to a nice little color tweaking PR. It's like eating dessert, but in PR form.

In this case, I polished up the mixin that colors Gutenberg according to your WordPress admin color scheme. For most schemes this means that the colors are a tad more accurate, and that their 2nd spot colors are now registered as SCSS variables, even if not yet used.

However in the case of Sunrise and Midnight themes, this 2nd spot color now _is_ being used, to color the Switch. Specifically, we want the _activated_ switch to always have a bright color that stays away from "warning" hues like red or orange. In the case of the Midnight theme, the switch used to be red (dark pink), and in Sunrise it used to be orange. This PR switches those to use the new 2nd spot colors (these are traditionally used for the "notification" color in wp-admin, such as you have 1 plugin update). This makes them light blue and olive, respectively.

vishalkakadiya committed:
Fix typo in php doc block for gutenberg_can_edit_post() (#6439)

* Define  variable as global

* Revert change

* Fix small typo in php doc

* Revert unwanted change

jorgefilipecosta committed:
Used align hook in columns block. Fixed bug where align classes were not passed to the frontend. (#6364)

mkaz committed:
Split loading of API actions and filters to own files (#6327)

* Split API init from compat.php to compat-api.php

* Split API init into own file

* Combine register and compat api into one rest-api.php file

* Remove permalink_structure per #6319

* Move new rest api function for theme supports to rest-api.php load file

* Move new register routes to api loader file

adamsilverstein committed:
Add isDebounced to autocompleter docs (#6446)

samikeijonen committed:
Add drop cap help text in paragraph block (#6415)

jorgefilipecosta committed:
Used IconButton on breadcrumbs to increase accessibility. (#6374)

This commit replaces the usage of Button + Dashicon with IconButton in order to increase accessibility as IconButton label is used as aria-label.
It looks like uid prop was passed to dashicon by mistake as the component does not do anything with this prop.

youknowriad committed:
Build Tools: Fix the package plugin script (#6463)

vishalkakadiya committed:
Makes + icon showing consistent across all library states

Props @vishalkakadiya.
Fixes #6090

ellatrix committed:
Raw handling: skip inline shortcodes (#6329)

* Skip inline shortcodes

* If the shortcode contains HTML, convert to block

jorgefilipecosta committed:
Implemented color class mechanism in button. (#6422)

danielbachhuber committed:
Switch to `rest_get_server()` for compat with trunk (#6474)

pento committed:
Expose Gutenberg Data Format version in the REST API (#6436)

* Add the block format version to the REST API post resource

* Add the block_format to the post content response.

* Add todo marker for including on schema

* Fix PHP 5.2 incompatibility

* Ensure this is an array to handle `isset()` behavior in 5.2

afercia committed:
Fix arrow navigation in the Shared Block more options menu. (#6483)

jorgefilipecosta commented on lib/client-assets.php:
```diff
@@ -929,7 +929,7 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	$gutenberg_theme_support = get_theme_support( 'gutenberg' );
 	$align_wide              = get_theme_support( 'align-wide' );
-	$color_palette           = get_theme_support( 'editor-color-palette' );
+	$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );
```
I think this may be breaking back compatibility with the existing version:
``` 
add_theme_support( 'editor-color-palette',
 	'#fff'
 	'#f00'
);
```
Maybe we should handle some condition to be backcompatible.

webmandesign replied:
My question would be, is this required? Gutenberg is still in development and from what I can see the theme support declaration have already changed. (Though, I don't know if it was made backwards compatible.)

Also, is this backwards compatible too?:
```
add_theme_support( 'editor-color-palette',
	array(
		'name' => 'color_name_1',
		'color' => '#color1',
	),
	array(
		'name' => 'color_name_2',
		'color' => '#color2',
	)
);
```

If needed, I can surely, edit my code to incorporate backwards compatibility.

jorgefilipecosta replied:
Hi all,
Thank you for sharing your thoughts, and for this contributions.
When changing colors mechanism, we always made efforts to try to be back-compatible in this case being back-compatible does not look very complicated. I think If we add this block:
```php
	// Backcompat for Color Palette set as multiple parameters.
	if ( is_string( $color_palette ) || isset( $color_palette['color'] ) ) {
		$color_palette = get_theme_support( 'editor-color-palette' );
		wp_add_inline_script(
			'wp-edit-post',
			'console.warn( "' .
				__( 'Setting colors using multiple parameters is deprecated. Please pass a single parameter with an array of colors. See https://wordpress.org/gutenberg/handbook/extensibility/theme-support/ for details.', 'gutenberg' ) .
			'");'
		);
	}
```
Bellow `$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );``, it should work with both our old versions:
```
add_theme_support( 'editor-color-palette',
	array(
		'name => 'red'
		'color' => '#ff0',
	),
	array(
		'name => 'green'
		'color' => '#0f0',
	),
	array(
		'name => 'blue'
		'color' => '#00f',
	)
);
```
and
```
add_theme_support( 'editor-color-palette',
	'#f00',
	'#0f0',
	'#00f'
);
```

Would it be possible to rebase this branch so we can test with our latest changes, e.g., the mechanism to hide color palettes if we don't have colors to choose?

@mtias what are your thoughts? In your opinion, should we advance with this changes and set the color palette using a single array parameter instead of the current way (multiple parameters)?

pento committed:
Add Embed Preview support for classic embed providers (#6345)

Fixes #3499.

Co-Authored-By: Andrew Duthie <aduth@users.noreply.github.com>

jasmussen committed:
Fix regression with formatting button hover/focus style (#6493)

A recent fix to aria-disabled buttons (up mover on first block) caused a regression in formatting buttons. This PR fixes that.

jorgefilipecosta committed:
Used editorMediaUpload in Gallery files transform (multiple images drag&drop). (#6456)

Before we used mediaUpload directly this created a bug. The images uploaded did not contain the reference to the current post where they were uploaded.

youknowriad committed:
Try adding a purify Higher Order Component (#6313)

* Try adding a purify Higher Order Component

* Rename purify => pure

* Unit test the pure HoC

gziolo committed:
Framework: Make sure block assets are always registered before wp-edit-post script (#6448)

* Framework: Make sure block assets are always registered before wp-edit-post script

* Framework: Use domReady to make sure that all blocks are properly registered

jorgefilipecosta committed:
Used color name in ColorPalette aria-label. (#6423)

This improves accessibility as it is useful for users to have a description of the color instead of just showing the color.

jorgefilipecosta committed:
Removed block transforms from the toolbars. (#6473)

jasmussen committed:
Fix select box regression. (#6350)

Fixes #6326.

This "undoes" a change to the select box styling, and polishes the focus style as well.

The regression was to introduce a padding change that conflicted with upstream select box styles. This essentially reverts that. This also polishes the focus styles a bit.

afercia committed:
Minor improvements for the permalink Copy to clipboard button (#6472)

* Permalink copy button minor improvements.

* Reset focus/hover styles, use aria-disabled, and update the label when the permalink is copied.

ellatrix committed:
Revert "Use nested blocks for quotes (#6054)" (#6501)

This reverts commit 853900d9ff38bc6f6ef0b71c70c30290783e3e3a.

gziolo committed:
Blocks: Extend BlockEdit context with name and use it for autocompleters (#6357)

* Blocks: Add block name to BlockEdit context

* Blocks: Use BlockEdit context and filter to add blocks autocompleter to Paragraph block

* Docs: Update Autocomplete document to include block name in examples

ajitbohra committed:
embed: fix translation string (#6354)

* embed: fix translation string

* Add translators note

jorgefilipecosta committed:
Improved accessibility of PanelColor. (#6428)

PanelColor displayed the color in a visual way as a square with the color, but for users with accessibility needs, we should also provide that information in a textual/descriptive way.

Rahmon committed:
Try prevent relative links if URL isn't fully typed out (#5917)

* Try to prevent relative links if URL isn't fully typed out

* Refactoring: move correctLink function to utils folder

* Update to use prependHTTP function by @wordpres/url package

* Remove correctLink function

* Framework: Update package-lock.json for url@1.1.0

aduth committed:
Data: Optimize withSelect shouldComponentUpdate (#6457)

tofumatt committed:
docs: Fix link to Repository Management (#6517)

jorgefilipecosta committed:
Refactor withColors HOC. (#6503)

The component now instead of passing an initializeColor prop, allows the users of the component to configure the mapping when instantiating the component.

Commit:
Update CONTRIBUTORS.md

jasmussen committed:
Fix 5919 in an alternate way (#6519)

In PR #6090 (much appreciated), #5919 was fixed. However this particular fix introduced a regression where the side plus was now visible right at the start, which it wasn't in 2.7.

This PR takes a different approach, fixing the same issue but without the regression.

ellatrix committed:
RichText: bind known positioned container (#6518)

afercia committed:
Simplify the BlockBreadcrumb component (#6507)

* Simplify BlockBreadcrumb component.

* Remove unused constant.

aduth committed:
Store: Reset change detection on post update (#6209)

* Store: Preserve change detection isDirty for ignored types

* Store: Reset change detection on post update

* Editor: Refactor dirtiness condition on pending transactions

* Testing: Add pressWithModifier E2E test utility

aduth committed:
Rich Text: Deprecate broad handling of events (#6286)

aduth committed:
Testing: Enforce array as Lodash path argument (#6247)

* Testing: Enforce _.get second parameter as array

* Testing: Enforce path argument as array universally

danielbachhuber committed:
Pre-load REST API index data to avoid FOMD (#6434)

Because `theme_supports` data present in the index is used in rendering UI, it'd be better to include it in the initial payload, instead of incurring a second API request.

youknowriad committed:
Cover Image Block: Remove the text alignment from the block inspector (#6497)

jasmussen committed:
Update theme extensibility documentation to include editor widths (#6531)

* Update theme extensibility documentation to include editor widths

This updates the extensibility document, and adds a section around loading editor styles. It gives examples for how to change the colors of the editor (which will look good once #6406 gets merged), as well as how to change the width of the editor. The last part fixes #1483.

CC: @maddisondesigns

* Simplify syntax a bit.

* Trim whitespace

youknowriad committed:
Cover Image: Hide inspector controls if no image is selected (#6498)

jorgefilipecosta committed:
Hide color pickers in paragraph and button if no colors are available. (#5570)

If the theme sets the colors palette to empty and disable custom colors, the color picker mechanism did not work correctly. It just showed a non-functional back color to choose from. Now we hide the color picker mechanism as in that case it is not possible to choose colors.

aduth committed:
Framework: Use @wordpress/is-shallow-equal for shallow equality (#6526)

* Framework: Use @wordpress/is-shallow-equal for shallow equality

* Framework: Replace all occurrences of dash in camelCaseDash

gziolo committed:
Editor: Make sure createInnerBlockList never updates when passed using context (#6420)

mtias committed:
Add "Spacer" block to create empty areas. (#6121)

Includes an aria-hidden attribute on save as this has no semantic meaning.

* Move to new core-blocks folder.
* Add fixtures.
* Improve setting controls with BaseControl and Panel.

aduth committed:
Rich Text: Determine emptiness by value (#5091)

* Rich Text: Determine emptiness by value

* Rich Text: Filter both sets of child nodes in split

* Rich Text: Consolidate isEmpty conditions

danielbachhuber committed:
Switch `post-author` component to use `/wp/v2/users/?who=authors` (#6515)

* Switch `post-author` component to use `/wp/v2/users/?who=authors`

* Ensure state is always an array, properly

* Update tests for 80d0500b2cca564303d5c14d891572420c0d4f1a

* Remove unused variable

* Remove unnecessary prop pass-through

* Ensure no duplicate authors are listed

* Use a "users" reducer combined with a "queries" sub state to map authors to users (#6522)

youknowriad committed:
Initial entities implementation (#6462)

* Initial models implementation

* Addd unit tests

* Use primaryKey instead of PK

* Fix selectors unit tests

* Move selectors/resolvers generators to the core-data/index.js

* Rename Model to Entity

* Rename primaryKey to just key

* Use the entity abstraction for media fetching

* Extract getMethodName helper

* Use lodash shortcut syntax

* Factorize the resovers/selectors generation

* Optimize entity reducer

* Fix small bug when generating selectors/resolvers

loicblascos committed:
Fix latest posts block orderby attribute typo (#6188)

ellatrix committed:
Add paste schema (fix various issues, simplify) (#5966)

danielbachhuber committed:
Disable link suggestions when value is URL (#4649)

* Disable link suggestions when value is URL

* Use regex check used by WPLink TinyMCE plugin

* Load these variables into scope as constants

* Fix linting issue

* Correct the comment

* Consolidate duplicative statements

danielbachhuber committed:
Further harden `who=authors` check by author support for post type (#6548)

pento committed:
Upgrade react-datepicker to 1.4.1 (#6550)

pento committed:
Upgrade showdown to 1.8.6 (#6551)

afercia committed:
Fix the clipboard button as IconButton usage. (#6532)

* Fix the clipboard button as IconButton usage.

* Better fix for the ClipboardButton.

* Let the icon be passed via additional props.

* No need to explicitly pass the icon prop.

jorgefilipecosta committed:
Add nested settings mechanism with a template configuration and allowedBlock blocks logic. (#6546)

This commit makes easy to set settings that configure how a given nested area behaves. For now, the only settings implemented are allowedBlocks and template.
"allowedBlocks" restricts what blocks we can nest inside a nesting area.
"template" defines a set of blocks that are "prefilled" inside a nesting area, using the same syntax of CPT templates.

danielbachhuber committed:
Only display featured image UI when theme supports it too (#6541)

* Only display featured image UI when theme supports it too

* JSX formatting style

youknowriad committed:
Blocks: Hide the movers and the block menu when typing (#6564)

aduth committed:
Framework: Drop deprecations slated for 2.8 removal (#6559)

jorgefilipecosta committed:
Revert "Removed block transforms from the toolbars. #6473" (#6566)

We decided that transforms should be a primary action, and we will iterate further to have transforms fit nicely in the toolbars.

jorgefilipecosta committed:
Refactor blocks->PanelColor, to avoid the need for the colorName prop (#6543)

With the creation of a PanelColor in blocks, there is no need to receive the colorName anymore as the component already has the color context available and can get the colorName from it.
This change makes the usage of the component easier. Existing core blocks that make use of the component were updated.

ellatrix committed:
Restore priority on embed block for raw transforming (#6572)

* Restore priority on embed block for raw transforming

* Add priority to paragraph instead of embed

mtias committed:
Update version to 2.8.0. (#6568)

tofumatt committed:
fix: Remove block alignment from paragraph block (fix #6476) (#6511)

* fix: Remove block alignment from paragraph block (fix #6476)
* chore: Deprecate width attribute from paragraph

hypest committed:
Port the core/code block to be used by the ReactNative mobile GB app (#5816)

* Android specific files for the Code block

* CSS related imports are suppressed
* classnames are left in only to minimize diff among the files but do not play a role, nor style the elements

* Breaking change: Don't override the default JSX translation

So far, that's the only way I found to have the android-specific jest tests succeed.

* Add the platform specific files for iOS as well

* Set TextInput to multiline for Code block

* Hide the default TextInput underline on Android

* Expose registerBlockType, getBlockType to the RN app

* Use .native.js for common native code

* Correct configuration for the babel plugins settings

* transform-react-jsx for the production env as well

* No need for pragma override

Will define a `wp` global via RN and reuse the `createElement()` from `@gutenberg/component`.

* Ignore mobile native files when linting

* Revert "Ignore mobile native files when linting"

This reverts commit a3d5ab4af34941e5afb12eba3b620dce565a78e6.

* Comment out ununsed imports

* Fix codestyle

* Import is compatible anyway, uncomment it

* Remove the React import. The RN project auto-imports it.

* Limit native code to the edit function alone

* Expose getBlockContent to RN

* RN app tests need createElement exported

* Export serialize from serializer

* Load native version of plain-text's CSS module

* Make it clear that RN uses styles, not classnames

* Use the i18n package from @wordpress

* Remove some commented out code

* Utils are not currently used on native

* 'Border' property not supported in shorthand form

* Use hypens in css class names as traditionally

* remove unused code

* No need for platform-specific factory.js

* Move native files after update from master

* Codestyle fix

* GB isn't using Prettier so, no need for the pragma

* Comment on the need to import the `edit` function

* Export the edit function as default

* Remove reduntant line comment

* Import block using the new pattern

* Add a comment about styling

* Use object property shorthand for `edit`

youknowriad committed:
Remove the document outline from the sidebar (#6557)

pento committed:
Fix a PHP Notice in REST API responses (#6576)

* Remove is_array() check from gutenberg_add_block_format_to_post_content()

* Add back backward compatible check

ehg committed:
Update plugin-compat.php (#6535)

ocean90 committed:
MediaUpload: Add documentation for `title` and `modalClass` props. (#6502)

ellatrix committed:
Add remaining editable shortcuts (#4411)

* Add remaining editable shortcuts
* Add shortcuts to formatting control tooltips
* Move shortcut text in tooltips
* Adjust tooltip styles
* feat: Add Keyboard Shortcut text utility
* chore: Improve Mac keycodes by using names instead of just glyphs

ellatrix committed:
Fix caretRangeFromPoint for Firefox (#6598)

tofumatt committed:
docs: Copy edits to theme extensibility (#6601)

Improve theme extensibility docs with copy edits; simplify changes made in #6531.

pento committed:
Update dom-react to 2.2.1 (#6610)

amdrew committed:
Fix spelling error in post-featured-image/index.js (#6609)

+ Other rormatting

ellatrix committed:
Add shortcut tooltips for main toolbar (#6605)

* Add shortcut tooltips for main toolbar

* Reorganise keycode file

* Add rawShortcut tests

* Polish

* No longer needed to omit keyboardShortcut

* Use rawShortcut everywhere

gziolo committed:
Blocks: Add automatic handling of focus for RichText component (#6419)

* Blocks: Add automatic handling of focus for RichText component

* Core blocks: Add automatic handling for focus for Quote and Pullquote blocks

* Blocks: Show controls for newly created block

* Blocks: Use method shorthand instead of arrow functions

* Blocks: Use onFocus on div to set focused element

* Blocks: Address review feedback

* Core Blocks: Remove obsolete `isSelected` usage from Spacer block

ryanwelcher committed:
Add SlotFill in Status & Availability panel to match existing hook (#6300)

* Introduce fill slot for the Status & Availbilty panel

* lint fixes

* Changing import statements to use the @wordpress package

* Fix to remove duplicate import

* Removing extra whitespace

* Fixes typo

* Rename slot to PluginPostStatusInfo

* Structural changes and only exporting the Fill to the public api

* Lint fixes

* Edit Post: Use PluginPostStatusInfo alias for Fill

* Edit Post: Fix export for the PluginPostStatusInfo

* Adding documentation notes

* Fixes typo in SlotFills

* Changes the docs

* Edit Post: Update documentation for plugins

* Edit Post: Small tweak with inline comments

gziolo committed:
Build: Introduce the common `build` folder to be used by all modules (#6584)

* Build: Put all build files into build top folder

* Framework: Move a11y and dom-ready to their own modules

* Edit Post: Move domReady logic to client-assets

* Make phpcs happy by adding dot at the end of comment

* Make Eslint happy by removing no longer used domReady import

youknowriad committed:
Framework: Move components from the blocks to the editor module (#6521)

* Framework: Move reusable block components from the blocks module to the editor module

* Add deprecation message for wp.blocks.*

* Update documentation to match the code

* Fix unit tests

* More deprecations

* Fix e2e tests

* Fix deprecated functions

* Editor: Update class name for RN files

* Use proper import statements for deprecated block components

* Ensure the editor module is loaded before third party plugins

* Tweak editor/components react-native version

* Removing debugging code

danielbachhuber committed:
Use `targetSchema` of JSON Hyper Schema to communicate sticky action (#6529)

* Use `targetSchema` of JSON Hyper Schema to communicate sticky action

Because WordPress' capabilities system is filterable, we need to execute
the capabilities before we know whether the current user can perform the
action. Fortunately, `targetSchema` supports exactly this use-case.

* Ensure links are properly formatted

* Avoid re-renders when the current post changes

* Technically more correct test

* Update tests for 3e591c50d08c3128745c44cb7c23d8b8ac991c5f

* Only include `wp:action-sticky` for `context=edit`; move description

* Move the description text from the targetSchema to the link title. (#6613)

pento committed:
Allow resetting the permalink by saving it as empty (#6615)

Fixes #6603.

aduth committed:
Components: Implement Button as assigning ref via forwardRef (#6527)

* Element: Expose forwardRef as export from wp.element

* Components: Implement Button as assigning ref via forwardRef

youknowriad committed:
Table of content: Add a max-height (#6586)

* Table of content: Add a max-height

* Add box shadow to indicate that the popover can be scrollable

hypest committed:
Port the core/more block to ReactNative (#6626)

* Register the _more_ block in blocks/library

* Move the edit function to its own file

* Embryo of a RN more tag

* Add read only text to the more tag

* Add PlainText input to the more block

* Remove the 100% width on the plain-text style

* Fix styling on the core/more block

* Move styles to .scss file

* Fix i18n import

* Remove font styling for the more block marker

* Replace underscores by hyphens in css names

* editor/index.js needs nativization

* Fix eslint issues

* Core blocks: Rename component to follow pattern

* Follow naming pattern in `code`, `more` edit modules

* Core blocks: Remove empty line from Code edit

ellatrix committed:
Fix multi selection with arrows + shift (#6606)

gziolo committed:
Core blocks: Extract edit to their own file - part 1 (#6637)

* Core blocks: Extract edit to their own file - part 1

* Own file for heading block's edit function

aduth committed:
Writing Flow: Consider tabbable edge if no adjacent tabbable (#6542)

Commit:
Adds better descriptions for all blocks (#6624)

Iterates on descriptions and audits them all as a result.

Props @michelleweber. Fixes #6623

tofumatt committed:
feat: Show block remove button for empty (paragraph) block (#6589)

Fix #6579

ntwb committed:
Use npm v6 is the minimum required version (#6629)

gziolo committed:
Core blocks: Refactor blocks to use edit file - part 2 (#6641)

noisysocks committed:
Update Notice README (#6653)

`content` is no longer a property as the content of the notice is defined by the children of the component.

From https://wordpress.slack.com/archives/C02QB2JS7/p1525848612000302

jorgefilipecosta committed:
Add FontSizePicker component and refactor paragraph block to use it. (#6618)

This change makes font size UI generic and allows other blocks to take advantage of the same UI used in the paragraph.
Other changes will follow that will abstract other font size logic from paragraph (not just the UI).
Themes will also be able to configure the font sizes. The future changes will use this work.

webmandesign committed:
Merge branch 'update/theme-support-color-palette' of https://github.com/webmandesign/gutenberg into update/theme-support-color-palette

webmandesign committed:
Backcompat for Color Palette set as multiple parameters.

Thanks to @jorgefilipecosta https://github.com/WordPress/gutenberg/pull/6437#discussion_r186741194

webmandesign commented:
Hi guys,
Just wanted to follow up on this. Wouldn't it be better if we close this PR and I will create a new one instead? This one seems to be messed up by all the code updates I think.

webmandesign commented:
I've went ahead and created a new, mess-less pull request containing all the code discussed here. #7025
Closing this one so.
